src/frontend/src/handler/handle_privilege.rs

```diff
@@ -136,6 +136,20 @@ fn make_prost_privilege(
                 grant_objs.push(PbObject::SourceId(source.id));
             }
         }
+        GrantObjects::Sinks(sinks) => {
+            let db_name = session.database();
+            let search_path = session.config().search_path();
+            let user_name = &session.auth_context().user_name;
+
+            for name in sinks {
+                let (schema_name, sink_name) =
+                    Binder::resolve_schema_qualified_name(db_name, name)?;
```

Comment:
I assume we just resolve the schema based on the session schema here?

Reply:
Nope, it only resolve the schema if user specified it. For example: `GRANT SELECT ON SINK schema.sink_name TO user`. If user doesn't specify it, the schema will be resolved by the session search path later.